sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/FlatEncoderSuite.scala

```diff
@@ -82,6 +82,16 @@ class FlatEncoderSuite extends ExpressionEncoderSuite {
     new NotJavaSerializable(15),
     Encoders.kryo[NotJavaSerializable].asInstanceOf[ExpressionEncoder[NotJavaSerializable]],
     "kryo object serialization")
+
+  // Java encoders
+  encodeDecodeTest(
+    "hello",
+    Encoders.genericJava[String].asInstanceOf[ExpressionEncoder[String]],
```

Comment:
I'd consider using `encoderFor(...)` here just to consolidate the logic about what the internal encoder type is.